main.js

```diff
@@ -142,26 +142,42 @@ ipcMain.on('installAddon', (e, addonObj) => {
   installAddon(addonObj, configObj.addonDir)
     .then((newAddon) => {
       installedAddonsDict[newAddon.name] = newAddon
-      saveToAddonList(configObj, installedAddonsDict)
+      //saveToAddonList(configObj, installedAddonsDict)
+      integrityCheck(installedAddonsDict, configObj)
+    })
+})
+
+// update addon listener, fetches new version number and downloads the addon
+ipcMain.on('installUpdate', (e, addonObj) => {
+  console.log('Received request to update addon ' + addonObj.name)
+  const URLObj = checkWhichHost(addonObj.URL)
+  parseAddonDetails(URLObj).then(addonObj => {
+    installAddon(addonObj, configObj.addonDir)
+    .then((newAddon) => {
+      installedAddonsDict[newAddon.name] = newAddon
       integrityCheck(installedAddonsDict, configObj)
     })
+  })
 })
 
-// updateAddon() listener
+// checkAddonUpdate() listener
 ipcMain.on('checkAddonUpdate', (e, addonObj) => {
-  console.log('Received request to check addon for updates')
-  checkUpdate(addonObj).then(resultObj => {
-    installedAddonsDict[addonObj].version = resultObj.version
-    installedAddonsDict[addonObj].status = resultObj.status
-    saveToAddonList(configObj, installedAddonsDict)
+  console.log(`Received request to check ${addonObj.name} for updates`)
+  checkUpdate(addonObj).then(updateStatus => {
+    if (installedAddonsDict[addonObj.name].status !== updateStatus) {
+      console.log("here")
```

Comment:
remove this the bottom console log is more verbose, also if youre using the console to debug try using debuggers in the future (vscode has one built in that can attach to processes)